src/test/groovy/graphql/execution/reactive/CompletionStageMappingPublisherTest.groovy

```diff
@@ -119,4 +120,38 @@ class CompletionStageMappingPublisherTest extends Specification {
 
     }
 
+
+    def "asynchronous mapping works with completion"() {
```

Comment:
I did real TDD on this test.  I write it first with async delay and saw the problem consistently happen.

I then fixed the code! :)